cartographer/cloud/internal/local_trajectory_uploader.cc

```diff
@@ -151,6 +156,21 @@ void LocalTrajectoryUploader::Shutdown() {
 }
 
 void LocalTrajectoryUploader::TryRecovery() {
+  auto get_channel_state = [this]() {
+    return client_channel_->GetState(false /* try_to_connect */);
+  };
+  if (get_channel_state() != grpc_connectivity_state::GRPC_CHANNEL_READY) {
```

Comment:
I find it easier to read with just `client_channel_->GetState(false /* try_to_connect */)`, then you can drop get_channel_state